### DISCUSSION TITLE
Search mode after v1.0.0 - how does it work?

### DISCUSSION THREAD

**Discussion by gschwim** [mason-org/mason.nvim#1432] [Q&A] [answered]
Hi!

I'm loving Mason and have been using it since late last year.

I just updated to 1.6.1 and found the usual "search" mode in the Mason UI (:Mason) no longer works the same. Instead of using '/' and typing my query, then being able to step through the search by repeatedly hitting '/' and <enter>, I'm put in "search mode" that shows extra information alongside each lsp package, but I can only search to the first hit. Hitting '/' then <enter> again exits search mode and leaves me at the first result.

I've tried to look through the docs but can't seem to locate any information about how the new search mode works. Maybe I've missed it somewhere?

Rolling back to v1.0.0 gets it working as previously. Moving [one commit ahead](https://github.com/williamboman/mason.nvim/commit/a18c031c72a3c7576ba5dc60ee30de8290c8757c) of that release seems to implement the behavior I'm seeing above.

Question - is there any method to use this new search mode to find things beyond the first result? Is this maybe a bug?

Thanks!
schwim

**Comment by williamboman** [comment]
Hello! So I'm not aware of using `/` to go the next search result is common practice or if it does something different than `n`/`N`, which I'd argue is the common way of traversing search entries in the buffer. I guess it could be considered a bug because the search mode logic doesn't consider `last-pattern` but only the command-line value, will see if it'd make sense to fix this without weird effects.

**Reply by williamboman** [reply, marked as answer]
Makes sense: https://github.com/williamboman/mason.nvim/pull/1445

**Reply by gschwim** [reply]
n / N works as you suggested. TBH I wasn't aware of that method, have been using / and ? to step forward and back forever. No need to fix unless it's causing issues at a larger scale.

Thanks!
schwim

**Comment by gschwim** [comment]
#1445 -> 1.6.2 fixes it. Nice work!
